### PR TITLE
Bjorncs/logserver rpc preparations 

### PR DESCRIPTION
**Pull request by bjorncs**
FYI @toregge @arnej27959

### CODE REVIEW

File: vespalog/src/main/java/com/yahoo/log/VespaFormat.java [hunk 1]
```diff
@@ -193,4 +200,14 @@ public static void formatException (Throwable t, StringBuilder sbuf) {
         sbuf.append(" nesting=").append(depth);
     }
 
+    static String formatThreadProcess(long processId, long threadId) {
+        if (processId == 0) {
+            return "";
```
Comment (arnej27959) [inline vespalog/src/main/java/com/yahoo/log/VespaFormat.java]: pretty sure it should be "-" to signify "unknown" here

File: vespalog/src/main/java/com/yahoo/log/LogMessage.java [hunk 2]
```diff
@@ -85,19 +109,37 @@ public static LogMessage parseNativeFormat(String msg) throws InvalidLogFormatEx
...
+        if (slashIndex == -1) {
+            return Long.parseLong(threadProcess);
+        }
+        return Long.parseLong(threadProcess.substring(0, slashIndex));
+    }
+
+    private static long parseThreadId(String threadProcess) {
+        int slashIndex = threadProcess.indexOf('/');
+        if (slashIndex == -1) {
+            return -1;
```
Comment (arnej27959) [inline vespalog/src/main/java/com/yahoo/log/LogMessage.java]: the format method thinks "0" means unknown but here it's "-1" that means unknown

**Review by geirst — APPROVED** [review]
👍

File: vespalog/src/main/java/com/yahoo/log/VespaFormat.java (hunk 1 above)
Comment (bjorncs) [reply vespalog/src/main/java/com/yahoo/log/VespaFormat.java]: Changed the implementation to assume that process id is always present

**Comment by bjorncs** [comment]
@geirst PTAL

File: logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    private final Executor executor = Executors.newSingleThreadExecutor();
+    private final LogDispatcher logDispatcher;
+    private final Method method;
+
+    ArchiveLogMessagesMethod(LogDispatcher logDispatcher) {
+        this.logDispatcher = logDispatcher;
+        this.method = new Method(METHOD_NAME, "bix", "bix", this::log)
+                .methodDesc("Archive log messages")
+                .paramDesc(0, "compressionType", "Compression type (0=raw)")
+                .paramDesc(1, "uncompressedSize", "Uncompressed size (0 if compression type 'raw')")
```
Comment (geirst) [inline logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java]: The code further down requires uncompressedSize == 'data length', so the comment should be updated.

File: logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                .paramDesc(2, "logRequest", "Log request encoded with protobuf")
+                .returnDesc(0, "compressionType", "Compression type (0=raw)")
+                .returnDesc(1, "uncompressedSize", "Uncompressed size (0 if compression type 'raw')")
```
Comment (geirst) [inline logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java]: The code further down sets this equal to 'data length', so the comment should be updated.

File: logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                    return;
+                }
+                logDispatcher.handle(ProtobufSerialization.fromLogRequest(logRequestPayload));
+                rpcRequest.returnValues().add(new Int8Value((byte)0));
+                byte[] responsePayload = ProtobufSerialization.toLogResponse();
+                rpcRequest.returnValues().add(new Int32Value(responsePayload.length));
+                rpcRequest.returnValues().add(new DataValue(responsePayload));
+                rpcRequest.returnRequest();
+            } catch (Exception e) {
+                log.log(Level.WARNING, e, () -> "Failed to handle log request: " + e.getMessage());
```
Comment (geirst) [inline logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java]: I think in this case we should setError() on the rpc request and return it?

**Review by geirst — APPROVED** [review]
👍 Some comments to consider.

**Comment by geirst** [comment]
@havardpe please also review commit bbc7aec "Add rpc interface to logserver"

File: logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java (hunk 3 above)
Comment (bjorncs) [reply logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java]: Fixed

File: logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java (hunk 4 above)
Comment (bjorncs) [reply logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java]: Fixed

File: logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java (hunk 5 above)
Comment (bjorncs) [reply logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java]: Agree - see new commits.

**Comment by bjorncs** [comment]
The 144f195 commit is conflicting with master - removing it. Will reappear in new PR.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**